src/main/java/it/eng/idsa/dataapp/util/MessageUtil.java

```diff
@@ -77,4 +163,48 @@ private String createContractAgreement(Path dataLakeDirectory) {
 		}
 		return contractAgreement;
 	}
+	
+	private Connector getSelfDescription() {
+		URI eccURI = null;
+		try {
+			eccURI = new URI(eccProperties.getRESTprotocol(), null, eccProperties.getHost(), eccProperties.getRESTport(), null, null, null);
+		} catch (URISyntaxException e) {
+			logger.error("Could not create URI for Self Description request.", e);
+		}
+
+		try {
+			return new Serializer().deserialize(restTemplate.getForObject(eccURI, String.class), Connector.class);
```

Comment:
Could we split those 2 commands in 2 lines, and add some logging, so that someone is not correct, we know what went wrong?
Will serializer work correct if rest call fails?

Can we wrap also above URI create to this catch?

Reply:
I will do that. If the call fails the deserialization will fail too, doesn't matter  if we get null or something else, unless the response is an instance of Connector in String representation.